util/chunk/chunk.go

```diff
@@ -66,6 +66,111 @@ func New(fields []*types.FieldType, cap, maxChunkSize int) *Chunk {
 	return chk
 }
 
+// WideNew create a new chunk for wide fields table.
+func WideNew(fields []*types.FieldType, cap, maxChunkSize int) *Chunk {
+	chk := new(Chunk)
+	chk.columns = make([]*column, len(fields))
+	chk.capacity = mathutil.Min(cap, maxChunkSize)
+	colMem := make([]column, len(fields))
+	var capInfo capInfo
+	for i := 0; i < len(fields); i++ {
+		chk.columns[i] = &colMem[i]
+		f := fields[i]
+		elemLen := getFixedLen(f)
+		if elemLen == varElemLen {
+			capInfo.add(preCalVarLenColumn(chk.capacity))
+		} else {
+			capInfo.add(preCalFixedLenColumn(elemLen, chk.capacity))
+		}
+	}
+
+	span := chunkSpan{
+		elemBuf:    make([]byte, capInfo.elemBuf),
+		offsets:    make([]int32, capInfo.offsets),
+		data:       make([]byte, capInfo.fixedColData),
+		nullBitmap: make([]byte, capInfo.nullBitmap),
+	}
```

Comment:
this also can more effective with 

```
	span := chunkSpan{
 		byteBuf:    make([]byte, byteTotal),
 		int32Buf:    make([]int32, int32Total),
 	}
```

but will meet bug with some executor impl return more data than `cap` assued.